evap/rewards/views.py

```diff
@@ -22,28 +23,58 @@
     RewardPointRedemptionEvent,
     SemesterActivation,
 )
-from evap.rewards.tools import grant_eligible_reward_points_for_semester, reward_points_of_user, save_redemptions
+from evap.rewards.tools import (
+    grant_eligible_reward_points_for_semester,
+    redeemed_points_of_user,
+    reward_points_of_user,
+    save_redemptions,
+)
 from evap.staff.views import semester_view
 
 
+@transaction.atomic
+def check_consistent_previous_redemption_counts(request):
+    list(request.user.reward_point_grantings.select_for_update())
+    list(request.user.reward_point_redemptions.select_for_update())
+    try:
+        reward = int(request.POST["previous_reward_points"])
+        redeemed = int(request.POST["previous_redeemed_points"])
+    except (KeyError, ValueError, TypeError) as e:
+        raise BadRequest("Invalid redeemed-points or left-points field in redemption request") from e
+    return reward == reward_points_of_user(request.user) and redeemed == redeemed_points_of_user(request.user)
```

Comment:
Could we just add those lines to `save_redemptions`? I think it may reduce code complexity (and we can't forget the call). Or do we have cases where we want to call `save_redemptions` without first checking the validity?

Reply:
Refactored it now, introducing an OutdatedRedemptionData-Exception, thrown in save_redemptions and catched in redeem_reward_points. The fact that we showed the message as an error and that we want to change the status to 409, forced me to introduce some changes to the structure of redeem_reward_points (function returns the desired status now)